cli/internal/packager/deploy.go

```diff
@@ -178,3 +185,69 @@ func deployComponents(tempPath componentPaths, assets config.ZarfComponent) {
 		git.PushAllDirectories(tempPath.repos)
 	}
 }
+
+// If privded package is a URL download it to a temp directory
+func HandleIfURL(packagePath string, shasum string, insecureDeploy bool) string {
+	// Check if the user gave us a remote package
+	providedURL, err := url.Parse(packagePath)
+	if err != nil || providedURL.Scheme == "" || providedURL.Host == "" {
+		logrus.WithField("archive", packagePath).Debug("The package provided is not a remote package.")
+		return packagePath
+	}
+
+	if !insecureDeploy && shasum == "" {
+		logrus.Fatal("When deploying a remote package you must provide either a `--shasum` or the `--insecure` flag. Neither were provided.")
+	}
+
+	// Check the extension on the package is what we expect
+	if !isValidFileExtension(providedURL.Path) {
+		logrus.Fatalf("The URL provided (%s) had an unrecognized file extension.\n", providedURL.String())
```

Comment:
maybe instead of "unrecognized extension" we could say something like, "only xx extensions are permitted".  Later on we may want to consider not using file extensions and instead inspect file type as that's more reliable.  Something like https://github.com/h2non/filetype might work.

Reply:
👍 I'll add that to the list of potential new tasks. I really like that idea.